clang/runtime/dpct-rt/include/dpct/dpl_utils.hpp

```diff
@@ -20,6 +20,7 @@
 #include "dpl_extras/algorithm.h"
 #include "dpl_extras/numeric.h"
 #include "dpl_extras/iterators.h"
+#include "dpl_extras/iterator_adaptor.h"
```

Comment:
I'd prefer to include this header file only when needed,  then not all thrust migration need boost installed.
 
option1 may be: 
change L23 to :  
#ifdef ITERATOR_ADAPTOR_REQUIRED 
#include "dpl_extras/iterator_adaptor.h"
#endif  

option2:  remove L23,  have DPCT to generate "#include "dpl_extras/iterator_adaptor.h" on demand.

Reply:
I agree, it should only be included when needed.  I think option 1 looks better to me.  I assume that the migration code would populate `ITERATOR_ADAPTOR_REQUIRED` if necessary then?

We don't have current plans to iterator_adaptor in other dpl_extras header files, however I can't say definitively it will never happen.

Reply:
>>>I assume that the migration code would populate ITERATOR_ADAPTOR_REQUIRED if necessary then? 
Yes.

Reply:
@danhoeflinger  This PR needs to be rebased to the latest SYCLomatic branch.

Reply:
Sorry I missed this.  I just rebased it.